api/envoy/config/route/v3/route_components.proto

```diff
@@ -760,6 +760,30 @@ message RouteAction {
     ConnectConfig connect_config = 3;
   }
 
+  message MaxStreamDuration {
+    // Specifies the maximum duration allowed for streams on the route. If not specified, the value
+    // from the :ref:`max_stream_duration
+    // <envoy_api_field_config.core.v3.HttpProtocolOptions.max_stream_duration>` field in
+    // :ref:`HttpConnectionManager.common_http_protocol_options
+    // <envoy_api_field_extensions.filters.network.http_connection_manager.v3.HttpConnectionManager.common_http_protocol_options>`
+    // is used.
+    google.protobuf.Duration max_stream_duration = 1;
+
+    // If present, and the request contains a `grpc-timeout` header
+    // <https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md>`_, use that value as the
+    // *max_stream_duration*, but limit the applied timeout to the maximum value specified here.
+    // If configured as 0, the maximum allowed timeout for gRPC requests is infinity.
```

Comment:
Maybe this should say something like "If set to 0, the grpc-timeout header will be used without modification."

Reply:
Makes sense.  Done.